### DISCUSSION TITLE
Strengthening the Release Process for Quarkiverse

### DISCUSSION THREAD

**Discussion by gastaldi** [quarkiverse/quarkiverse#284] [Team Posts]
Hello @quarkiverse/quarkiverse-members!

Due to https://quarkus.io/blog/quarkiverse-and-smallrye-new-release-process/, we have changed our release process internally. The organization's secrets were moved to the https://github.com/quarkiverse/quarkiverse-release repository, so the `release.yml` in your repository won't work anymore. The Maven central credentials were also rotated for extra security.
 
## What do I need to do? 

You'll find a PR titled `Split and Reuse the Release workflow` in your repository. If you haven't changed the original release workflow, merge it.   

## The workflow is merged. How do I trigger a new release?

As described in https://hub.quarkiverse.io/release/, the steps to perform a release haven't changed.

## I need help migrating my release workflow

Ping us in this thread and we'll be glad to help you.

**Comment by melloware** [comment]
I would just like to say I LOVE the new release process if for nothing else than it publishes the release to GitHub Releases which I always was doing manually 😄

**Reply by metacosm** [reply]
Ah, nice! Haven't tried it yet but good to know!

**Comment by CodeSimcoe** [comment]
Hello @gastaldi
For extension [quarkus-fx](https://github.com/quarkiverse/quarkus-fx) we need a DISPLAY to be present for tests, that's why `build.yml` makes use of xvfb :

```
      - name: Run Tests
        run: xvfb-run ./mvnw -B -ntp verify
```


Since the new release update, tests do fail on release

```
[INFO] 2024-10-18 06:43:54,153 ERROR [io.qua.fx.QuarkusFxApplication] (Thread-30) An exception occurred in Fx application launch: java.lang.UnsupportedOperationException: Unable to open DISPLAY
```

I bet `release-perform.yml` and `release-prepare.yml` shall be updated ?
Not really familiar with these pipelines, can you help ?

Also, currently the tests fails but the action doesn't until it times out.

Thanks

**Reply by gastaldi** [reply]
@CodeSimcoe I've created https://github.com/quarkiverse/quarkus-fx/pull/91 using the [prepare-release-headless.yml](https://github.com/quarkiverse/.github/commit/918ff94c1b412547936beda0e80b0a783d1116ef) I introduced in the .github repository. Let me know if that works for you

**Reply by CodeSimcoe** [reply]
It fails for now
```
(java:1942): dbind-WARNING **: 13:35:29.745: AT-SPI: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
2024-10-18 13:35:29,833 INFO  [io.quarkus] (main) quarkus-fx-deployment stopped in 0.005s
Error:  Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 3.362 s <<< FAILURE! -- in io.quarkiverse.fx.deployment.FxStartupTest
```

**Reply by gastaldi** [reply]
@CodeSimcoe I fixed the release by introducing [a `maven_executable` input](https://github.com/quarkiverse/quarkus-fx/blob/main/.github/workflows/release-prepare.yml#L19-L20).

Release 0.8.0 worked as a charm: https://github.com/quarkiverse/quarkus-fx/releases/tag/0.8.0

**Reply by CodeSimcoe** [reply]
Thanks a lot @gastaldi !

**Comment by turing85** [comment]
Hi!
@gastaldi and I just found something.
The new release process auto-generates release notes. This new process does not honor SemVer. So for example, if we release a version `3.6.0` and a version `3.5.1` afterwards (which we just did in `quarkus-artemis`), the later but "smaller" version (i.e. `3.5.1`) will be set as `latest`.

**Reply by gastaldi** [reply]
I created https://github.com/quarkiverse/quarkiverse-release/issues/7 to track this

**Comment by metacosm** [comment]
Another side-effect to this change: I have workflows that release snapshot versions of my extension. These do not work anymore because they were using the GPG key for artifact signing. Could we have such a snapshot release workflow in Quarkiverse, please?

**Reply by gastaldi** [reply]
Of course: here is an example: https://github.com/quarkiverse/quarkus-prettytime/blob/main/.github/workflows/deploy-snapshots.yml

It's also available when you create a new workflow here: https://github.com/quarkiverse/quarkus-operator-sdk/actions/new


![Screenshot_20241028_085925_Chrome](https://github.com/user-attachments/assets/3546977f-b7fa-4a3c-84b5-29dfa12351af)

**Reply by metacosm** [reply]
I assume that the snapshot deployment is only triggered if we set a version to something `-SNAPSHOT`?

**Reply by gastaldi** [reply]
Yes, that's correct, but the secret is in the `ref` parameter. In a normal deployment it would be the tag version

**Reply by metacosm** [reply]
Been having problems with this since today: the token creation has been failing, see for example: https://github.com/quarkiverse/quarkus-operator-sdk/actions/runs/11610645678/job/32330581819#step:13:15

**Reply by gastaldi** [reply]
Your workflow is missing `secrets: inherit`. Here is a PR fixing it: https://github.com/quarkiverse/quarkus-operator-sdk/pull/989

**Comment by metacosm** [comment]
It would also be useful to be able to pass maven arguments to the release process so that people can activate profiles, set properties, etc.

**Reply by gastaldi** [reply]
PRs are welcome 😉
